### PR TITLE
[CINFRA] Agency Supervision Refactor

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose
This PR tries to refactor the supervision code, without changing its behavior. Additionally
- unused data types have been removed
- an unused shutdown mechanism has been removed
- The network feature included the supervision.h and this caused the methods.cpp and utils.cpp to be rebuild every time on modifies the supervision.

### CODE REVIEW

File: arangod/Agency/Agent.h [hunk 1]
```diff
@@ -385,7 +386,7 @@ class Agent final : public arangodb::ServerThread<ArangodServer>,
   Constituent _constituent;
 
   /// @brief Cluster supervision module
-  Supervision _supervision;
+  std::unique_ptr<Supervision> _supervision;
```
Comment (markuspf) [inline arangod/Agency/Agent.h:389]: Out of interest: Why?

Comment (maierlars) [reply arangod/Agency/Agent.h:389]: This allows me to have this header included without making the implementation of Supervision visible. Not having it as a pointer would require every user of Agent to also include Supervision.h.

**Review by markuspf — APPROVED** [review]
LGTM

**Review by neunhoef — COMMENTED** [review]
LGTM

**Pull request merged**